File: Z-Apps/ClientApp/src/types/vocab.ts
export type vocab = {
    genreId: number;
    vocabId: number;
    hiragana: string;
    kanji: string;
    english: string;
    order: number;
};
export type vocabGenre = {
    genreId: number & { _vocabGenreId: never };
    genreName: string;
    order: number;
    youtube: string;
    released: boolean;
};
export function VocabGenreId(id: number) {
    return id as vocabGenre["genreId"];
}
export type vocabMergedGenre = {
    genreId: number & { _mergedVocabGenreId: never };
    genreName: string;
    order: number;
    youtube: string;
    released: boolean;
};
export function VocabMergedGenreId(id: number) {
    return id as vocabMergedGenre["genreId"];
}
export type sound = {
    audio: HTMLAudioElement;
    playable: boolean;
};

File: Z-Apps/Models/Vocab/VocabList/Vocab.cs
namespace Z_Apps.Models.VocabList
{
    public class Vocab
    {
        public int genreId { get; set; }
        public int vocabId { get; set; }
        public string kanji { get; set; }
        public string hiragana { get; set; }
        public string english { get; set; }
        public int order { get; set; }
    }
}